server/src/main/kotlin/eu/kanade/tachiyomi/source/LocalSource.kt

```diff
@@ -1,153 +1,170 @@
+import com.google.gson.JsonParser
+import eu.kanade.tachiyomi.source.model.Filter
 import eu.kanade.tachiyomi.source.model.FilterList
 import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.lang.compareToCaseInsensitiveNaturalOrder
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.insertAndGetId
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.kodein.di.DI
+import org.kodein.di.conf.global
+import org.kodein.di.instance
 import rx.Observable
+import suwayomi.tachidesk.manga.model.table.ExtensionTable
+import suwayomi.tachidesk.manga.model.table.SourceTable
+import suwayomi.tachidesk.server.ApplicationDirs
+import java.io.File
+import java.io.FileNotFoundException
+import java.net.URL
+import java.util.Locale
+import java.util.concurrent.TimeUnit
 
-// import com.github.junrar.Archive
-// import com.google.gson.JsonParser
-// import eu.kanade.tachiyomi.R
-// import eu.kanade.tachiyomi.source.model.Filter
-// import eu.kanade.tachiyomi.source.model.FilterList
-// import eu.kanade.tachiyomi.source.model.MangasPage
-// import eu.kanade.tachiyomi.source.model.Page
-// import eu.kanade.tachiyomi.source.model.SChapter
-// import eu.kanade.tachiyomi.source.model.SManga
-// import eu.kanade.tachiyomi.util.chapter.ChapterRecognition
-// import eu.kanade.tachiyomi.util.lang.compareToCaseInsensitiveNaturalOrder
-// import eu.kanade.tachiyomi.util.storage.DiskUtil
-// import eu.kanade.tachiyomi.util.storage.EpubFile
-// import eu.kanade.tachiyomi.util.system.ImageUtil
-// import rx.Observable
-// import timber.log.Timber
-// import java.io.File
-// import java.io.FileInputStream
-// import java.io.InputStream
-// import java.util.Locale
-// import java.util.concurrent.TimeUnit
-// import java.util.zip.ZipFile
-
-class LocalSource(private val context: Context) : CatalogueSource {
+class LocalSource(override val baseUrl: String = "") : HttpSource() {
     companion object {
         const val ID = 0L
-//        const val HELP_URL = "https://tachiyomi.org/help/guides/reading-local-manga/"
-//
-//        private const val COVER_NAME = "cover.jpg"
-//        private val SUPPORTED_ARCHIVE_TYPES = setOf("zip", "rar", "cbr", "cbz", "epub")
-//
-//        private val POPULAR_FILTERS = FilterList(OrderBy())
-//        private val LATEST_FILTERS = FilterList(OrderBy().apply { state = Filter.Sort.Selection(1, false) })
-//        private val LATEST_THRESHOLD = TimeUnit.MILLISECONDS.convert(7, TimeUnit.DAYS)
-//
+        const val LANG = "localsourcelang"
```

Comment:
Why not `all` or `other`?

Reply:
sources use both those langs and having a special lang makes categorizing it into a different section easier.